lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -1191,43 +1446,44 @@ private void ReadPersonFrom(XmlReader reader, SyndicationPerson result)
                         ((IDisposable)extWriter).Dispose();
                     }
                 }
-                reader.ReadEndElement();
+                await reader.ReadEndElementAsync();
             }
         }
-
-        private TextSyndicationContent ReadTextContentFrom(XmlReader reader, string context)
+        
+        private async Task<TextSyndicationContent> ReadTextContentFromAsync(XmlReaderWrapper reader, string context)
         {
-            return ReadTextContentFrom(reader, context, this.PreserveAttributeExtensions);
+            return await ReadTextContentFromAsync(reader, context, this.PreserveAttributeExtensions);
         }
-
-        private void WriteCategoriesTo(XmlWriter writer, Collection<SyndicationCategory> categories)
+        
+        private async Task WriteCategoriesToAsync(XmlWriter writer, Collection<SyndicationCategory> categories)
         {
+            XmlWriterWrapper wrappedWriter = XmlWriterWrapper.CreateFromWriter(writer);
             for (int i = 0; i < categories.Count; ++i)
             {
-                WriteCategory(writer, categories[i], this.Version);
+                await WriteCategoryAsync(wrappedWriter, categories[i], this.Version);
             }
         }
-
-        private void WriteFeed(XmlWriter writer)
+        
+        private async Task WriteFeed(XmlWriterWrapper writer)
```

Comment:
Rename to WriteFeedAsync
Remove async and await and return Task from WriteFeedToAsync